src/main/java/org/betonquest/betonquest/commands/QuestCommand.java

```diff
@@ -1813,6 +1829,135 @@ private Optional<List<String>> completeDownload(final String... args) {
         };
     }
 
+    /**
+     * Variables stuff.
+     */
+    @SuppressWarnings("PMD.NcssCount")
+    private void handleVariables(final CommandSender sender, final String... args) {
+        final Profile profile = getTargetProfile(sender, args);
+        if (profile == null) {
+            return;
+        }
+
+        final boolean isOnline = profile.getOnlineProfile().isPresent();
+        if (!isOnline) {
+            log.debug("Can't access variable data on offline player");
+            sendMessage(sender, "offline_invalid");
+            return;
+        }
+
+        if (args.length == 2) {
+            log.debug("Missing objective instruction string");
+            sendMessage(sender, "specify_objective");
+            return;
+        }
+
+        // get the objective
+        final ObjectiveID objectiveID;
+        try {
+            objectiveID = new ObjectiveID(null, args[2]);
+        } catch (final ObjectNotFoundException e) {
+            sendMessage(sender, "error", e.getMessage());
+            log.warn("Could not find objective: " + e.getMessage(), e);
+            return;
+        }
+        final Objective tmp = BetonQuest.getInstance().getObjective(objectiveID);
+        if (!(tmp instanceof final VariableObjective variableObjective)) {
+            if (tmp == null) {
+                log.debug("Missing objective instruction string");
+            } else {
+                log.debug(tmp.getLabel() + " is not a variable objective");
+            }
+            sendMessage(sender, "specify_objective");
+            return;
+        }
+        log.debug("Using variable objective " + variableObjective.getLabel());
+
+        // if there are no arguments then list player's objective's entries
+        if (args.length < 4 || "list".equalsIgnoreCase(args[3]) || "l".equalsIgnoreCase(args[3])) {
+            // check for actual values
+            final Map<String, String> properties = variableObjective.getProperties(profile);
+            if (properties == null) {
+                log.debug("No property for profile");
+                sendMessage(sender, "player_no_property");
+                return;
+            }
+            // display variable objective keys and values
+            log.debug("Listing keys and values");
+            final Predicate<String> shouldDisplay = createListFilter(args, 4, Function.identity());
+            sendMessage(sender, "player_variables", variableObjective.getLabel());
+            properties.entrySet().stream()
+                    .filter(entry -> shouldDisplay.test(entry.getKey()))
+                    .sorted((o1, o2) -> o1.getKey().compareToIgnoreCase(o2.getKey()))
+                    .forEach(entry -> sender.sendMessage("§b- " + entry.getKey() + "§e: §a" + entry.getValue()));
+            return;
+        }
+        switch (args[3].toLowerCase(Locale.ROOT)) {
+            case "set":
+            case "s":
```

Comment:
I am a bit confused why we use a switch here while in line 1877 we use equalsIgnoreCase. Should we not use the switch case also for list?
And one more question. How to delete a variable?

Reply:
I copied that part of the logic from the point event where no argument is the same as the list argument. I thought about breaking that and just use a new method for that so I could call it either before the length check and in the switch.

You delete a variable when not giving any argument (like the event with "") but I will clarify that in the command description.

Reply:
Hm, I don't like this implicit deletion, prevents the possibility of empty variables (although they make no sense atm) and I can already see the loads of support cases and bug tickets.

Even if the behavior is the same as a set empty for now I'd add a delete command.